src/components/UserSelector.tsx

```diff
@@ -1,35 +1,92 @@
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+
+type Props = {
+  users: User[];
+  selected: User | null;
+  getUsersPosts: (userId: number) => void;
+  onSelect: React.Dispatch<React.SetStateAction<User | null>>;
+  setIsCommentsListHidden: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  users,
+  selected,
+  onSelect,
+  getUsersPosts,
+  setIsCommentsListHidden,
+}) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const selectRef = useRef(null);
+
+  const handleSelectUser = (user: User) => {
+    onSelect(user);
+    setIsVisible(false);
+    getUsersPosts(user.id);
+    setIsCommentsListHidden(true);
+  };
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (event.target !== selectRef.current && isVisible) {
+      setIsVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('click', handleClickOutside);
+
+    return () => {
+      document.removeEventListener('click', handleClickOutside);
+    };
+  }, [isVisible]);
 
-export const UserSelector: React.FC = () => {
   return (
     <div
       data-cy="UserSelector"
       className="dropdown is-active"
+      ref={selectRef}
     >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setIsVisible(() => true)}
```

Comment:
no reason to use callback
```suggestion
          onClick={() => setIsVisible(true)}
```